cupy/_environment.py

```diff
@@ -2,35 +2,71 @@
 This file must not depend on any other CuPy modules.
 """
 
+import distutils
 import os
 import os.path
+import shutil
 
 
 _cuda_path = None
+_nvcc_path = None
 
 
 def get_cuda_path():
+    # Returns the CUDA installation path or None if not found.
     global _cuda_path
     if _cuda_path is None:
-        _cuda_path = os.getenv('CUDA_PATH', None)
-        if _cuda_path is not None:
-            return _cuda_path
-
-        for p in os.getenv('PATH', '').split(os.pathsep):
-            for cmd in ('nvcc', 'nvcc.exe'):
-                nvcc_path = os.path.join(p, cmd)
-                if not os.path.exists(nvcc_path):
-                    continue
-                nvcc_dir = os.path.dirname(os.path.abspath(nvcc_path))
-                _cuda_path = os.path.normpath(os.path.join(nvcc_dir, '..'))
-                return _cuda_path
-
-        if os.path.exists('/usr/local/cuda'):
-            _cuda_path = '/usr/local/cuda'
-
+        _cuda_path = _get_cuda_path()
     return _cuda_path
 
 
+def get_nvcc_path():
+    # Returns the path to the nvcc command or None if not found.
+    global _nvcc_path
+    if _nvcc_path is None:
+        _nvcc_path = _get_nvcc_path()
+    return _nvcc_path
+
+
+def _get_cuda_path():
+    # Use environment variable
+    cuda_path = os.environ.get('CUDA_PATH', '')  # Nvidia default on Windows
+    if os.path.exists(cuda_path):
+        return cuda_path
+
+    # Use nvcc path
+    nvcc_path = shutil.which('nvcc')
+    if nvcc_path is not None:
+        return os.path.dirname(os.path.dirname(nvcc_path))
+
+    # Use typical path
+    if os.path.exists('/usr/local/cuda'):
+        return '/usr/local/cuda'
+
+    return None
+
+
+def _get_nvcc_path():
+    # Honor the "NVCC" env var
+    nvcc_path = os.environ.get('NVCC', None)
+    if nvcc_path is not None:
+        return distutils.util.split_quoted(nvcc_path)
```

Comment:
This is an extra piece I added. Not sure if this is acceptable, but would be nice for users to be able to overwrite, given that `RawKernel` can use `nvcc` as the alternative backend.